### ISSUE TITLE
Inherit hadronic target to PineAPPL

### ISSUE THREAD

**Issue by felixhekhorn** [NNPDF/yadism#122]
in the PineAPPL generation we're hard coding at the moment the hadronic target: this might become a problem with NuTeV and CHORUS

https://github.com/N3PDF/yadism/blob/23c413de082d550b2161512b90a2157e2e69c3c2/src/yadism/output.py#L240

(the lepton is already correct:
https://github.com/N3PDF/yadism/blob/23c413de082d550b2161512b90a2157e2e69c3c2/src/yadism/output.py#L241
)

**Comment by felixhekhorn** [comment]
> > Yet another question: for the previous CHORUS dataset you set `initial_state_1` to `2212`, so although this is supposed to be an antineutrino-lead scattering you require a proton PDF. I suppose that means that all nuclear effects are taken care of inside yadism, is my understanding correct?
> 
> yes, inside yadism we're applying "level-0" nuclear corrections, i.e.
> 
>     * adjust the electric charge
> 
>     * swap u and d
> 
> 
> this is related to [N3PDF/yadism#122](https://github.com/N3PDF/yadism/issues/122)
> 
> I don't think this is blocking at the moment ...
> 
> However, since we're speaking about it, I have a naive question: even with the correct initial state we still need to apply the "level-0" nuclear corrections inside `yadism`, right?

If you're taking care of 'translating the proton to the nuclear target' then everything's fine as far as I understand it.

_Originally posted by @cschwan in https://github.com/NNPDF/runcards/issues/117#issuecomment-1012168872_

**Comment by felixhekhorn** [comment]
Let's move the discussion here, where it belongs ...

**Comment by felixhekhorn** [comment]
> If you're taking care of 'translating the proton to the nuclear target' then everything's fine as far as I understand it.

well, that's the question:
- do we need to do 1 thing: (apply "level-0" nuclear correction + use initial_state=p) or (no nuclear-correction + initial_state = Fe)
- or do we need to do 2 things: apply "level-0" nuclear correction + initial_state = Fe

**Comment by AleCandido** [comment]
This issue is getting old without nothing happening, nor going to happen soon.

@cschwan @felixhekhorn my personal point of view is:
- option 1 should be the default, not only because of PDF fitting (that is for sure relevant), but because it's easier you have proton PDFs and you're satisfied with level-0 corrections
- option 2 is always viable, without extra effort in `yadism`: you just need to pass a proton as a target in the runcard, and change later on the target inside the PineAPPL grid

and because of these two points, I'd say that we're fine with what we have. Anything more involved would be too much involved.

The only task we might consider (according to me) is to document this choice somewhere, spelling out the full recipe for option 2. Most likely `yadism` docs is still the best place, but if you have any other suggestion, please tell.

**Comment by cschwan** [comment]
@AleCandido I agree with you. There are also many more proton PDFs than nuclear ones, so that's yet another point in favour of option 1. Documenting it would be nice, if that's not already the case. For option 2 we'd need https://github.com/N3PDF/pineappl/issues/109.

**Comment by AleCandido** [comment]
@felixhekhorn and @cschwan we should reconsider this, and take action (I believe it should be a quick-fix in any case, so better attempting to close).
